lib/jobs/ipmi-job.js

```diff
@@ -184,20 +184,117 @@ function ipmiJobFactory(
     };
 
     /**
-     * Collect SEL entries list from IPMI
+     * gets the SEL information using the "sel get" ipmi command
      * @memberOf IpmiJob
      *
      * @param data
-     * @param count
+     * @param stringifiedEntries
      */
-    IpmiJob.prototype.collectIpmiSel = function(data, count) {
-        count = count || 25;
-        return ipmitool.sel(data.host, data.user, data.password, count)
-        .then(function (sel) {
-            return parser.parseSelData(sel);
-        });
+    IpmiJob.prototype.getSelEntries = function(data, stringifiedEntries) {
+        return ipmitool.sel(data.host, data.user, data.password,  stringifiedEntries)
+    };
+
+    /**
+     * gets the raw information of a single sel entry
+     * @memberOf IpmiJob
+     *
+     * @param data
+     * @param recordId
+     */
+    IpmiJob.prototype.getRawSelEntry = function(data, recordId) {
+        return ipmitool.readRawSel(data.host, data.user, data.password, recordId)
+    };
+
+    /**
+     * Collect SEL entries list from IPMI and add two properties ("Sensor Type Code" and "Event Type Code") to each sel entry
+     * @memberOf IpmiJob
+     *
+     * @param data
+     */
+    IpmiJob.prototype.collectIpmiSel = function(data) {
+        var self = this;
+        var stringifiedEntries = "";
+        var lastUpdatedSelDate;
+        var currentSelDate;
+        var lastReportedSelEntryID;
+        var totalEntries;
+        var lastUpdatedSelDate;
+        var sel;
+
+        return Promise.all([
+           self.collectIpmiSelInformation(data),
```

Comment:
I saw that the `sel get` call was querying a limited range of entries, but my reading of this code was that the initial call to `sel` from `collectIpmiSelInformation` was not limited to any range, but I could be totally reading the code wrong :)